creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1194,35 +1195,44 @@ class SnapbackSM {
    * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
    */
   async aggregateReconfigAndPotentialSyncOps(nodeUsers, unhealthyPeers) {
+    // Parallelize calling this._aggregateOpsWithQueriedSpIds on chunks of 500 nodeUsers at a time
+    const nodeUserBatches = _.chunk(
+      nodeUsers,
+      AGGREGATE_RECONFIG_AND_POTENTIAL_SYNC_OPS_BATCH_SIZE
+    )
+    const resultBatches = []
+    while (nodeUserBatches.length) {
+      const nodeUserBatch = nodeUserBatches.shift()
+      const resultBatch = await Promise.allSettled(
+        nodeUserBatch.map((nodeUser) =>
+          this._aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers)
+        )
+      )
+      resultBatches.push(resultBatch)
+    }
+
+    // Combine each batch's requiredUpdateReplicaSetOps and potentialSyncRequests
+    const results = _.flatten(resultBatches)
     let requiredUpdateReplicaSetOps = []
     let potentialSyncRequests = []
-
-    for (const nodeUser of nodeUsers) {
-      const { primarySpID, secondary1SpID, secondary2SpID } = nodeUser
-
-      // If these spIds are undefined, it means that the Discovery Node does not have the latest code to return
-      // these fields. Fallback to the original aggregation logic. If present, use the spIds to determine whether
-      // reconfigs/syncs are necessary.
-      let subsetReplicaOps, subsetSyncReqs
-      if (primarySpID && secondary1SpID && secondary2SpID) {
-        ;({
-          requiredUpdateReplicaSetOps: subsetReplicaOps,
-          potentialSyncRequests: subsetSyncReqs
-        } = await this._aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers))
-      } else {
-        // TODO: remove this else case once all the Discovery Nodes have upgraded and return the necessary replica set spIds
-        ;({
-          requiredUpdateReplicaSetOps: subsetReplicaOps,
-          potentialSyncRequests: subsetSyncReqs
-        } = await this._aggregateOpsWithoutQueriedSpIds(
-          nodeUser,
-          unhealthyPeers
-        ))
+    for (const promiseResult of results) {
+      // Skip failed promises
```

Comment:
can you document in which scenario the promise could fail and why it's ok to ignore?

Reply:
I actually don't know why it would fail -- haven't encountered it failing myself. I skip here because the previous code was just failing if any promise failed, so this was an improvement to continue and not abort the whole sync from one failed user. I'll add a log here so we can keep an eye out for unexpected failures

Reply:
ah yeah, i just looked it over. agreed, it should basically never fail
i think skipping + logging here is great (seems like a good place for a prometheus metric + alert maybe? 👀)

per the allSettled docs, do you think it may be valuable to check the `result.reason` field as well? seems like the `result.value` will be the error msg? i'm unsure tho
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#return_value

Reply:
whoops yeah good catch that should definitely be `reason` not `value`